src/App.jsx

```diff
@@ -14,33 +15,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState('');
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goodsFromServer].sort();
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortType('alphabet');
+    setIsReversed(false);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const handleSortByLength = () => {
+    const sortedGoods = [...goodsFromServer].sort(
+      (a, b) => a.length - b.length,
+    );
+
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortType('length');
+    setIsReversed(false);
+  };
+
+  const handleReverse = () => {
+    setGoods([...goods].reverse());
+    setIsReversed(!isReversed);
+  };
```

Comment:
The name of the method `handleReverse` doesn't start with a verb. According to the checklist, method's name should start with a verb. You can rename it to `reverseGoods` for example.